src/shim/preload_libraries.c

```diff
@@ -477,3 +480,70 @@ FILE *tmpfile(void) {
         free(name);
     return rv;
 }
+
+static void _convert_stat_to_stat64(struct stat* s, struct stat64* s64) {
+    memset(s64, 0, sizeof(*s64));
+
+    s64->st_dev = s->st_dev;
+    s64->st_ino = s->st_ino;
+    s64->st_nlink = s->st_nlink;
+    s64->st_mode = s->st_mode;
+    s64->st_uid = s->st_uid;
+    s64->st_gid = s->st_gid;
+    s64->st_rdev = s->st_rdev;
+    s64->st_size = s->st_size;
+    s64->st_blksize = s->st_blksize;
+    s64->st_blocks = s->st_blocks;
+    s64->st_atim = s->st_atim;
+    s64->st_mtim = s->st_mtim;
+    s64->st_ctim = s->st_ctim;
+}
+
+static void _convert_statfs_to_statfs64(struct statfs* s, struct statfs64* s64) {
+    memset(s64, 0, sizeof(*s64));
+
+    s64->f_type = s->f_type;
```

Comment:
Optional - might be worth a macro; as above.

Reply:
Done.